### PR TITLE
change FrozenSequence to boolean in solo machine

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

This will be a lot easier to do now since the solo machine was broken and we don't have to worry about backwards compatibility

closes: #36 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/ibc/lightclients/solomachine/v1/solomachine.proto [hunk 1]
```diff
@@ -16,7 +16,7 @@ message ClientState {
   // latest sequence of the client state
   uint64 sequence = 1;
   // frozen sequence of the solo machine
-  uint64         frozen_sequence = 2 [(gogoproto.moretags) = "yaml:\"frozen_sequence\""];
+  bool           is_frozen       = 2 [(gogoproto.moretags) = "yaml:\"is_frozen\""];
```
Comment (colin-axner) [inline proto/ibc/lightclients/solomachine/v1/solomachine.proto:19]: this is proto breaking. Should I make a v2? @AdityaSripal @fedekunze

Comment (colin-axner) [reply proto/ibc/lightclients/solomachine/v1/solomachine.proto:19]: Going to try to do a v2. I think it is good practice

Comment (colin-axner) [reply proto/ibc/lightclients/solomachine/v1/solomachine.proto:19]: Cool, I just changed the package from v1 -> v2

**Review by fedekunze — APPROVED** [review]
ACK

File: modules/light-clients/06-solomachine/types/client_state.go [hunk 2]
```diff
@@ -45,23 +45,18 @@ func (cs ClientState) GetLatestHeight() exported.Height {
...
-	if cs.FrozenSequence != 0 {
+	if cs.IsFrozen {
 		return exported.Frozen
 	}
 
 	return exported.Active
 }
 
-// IsFrozen returns true if the client is frozen.
-func (cs ClientState) IsFrozen() bool {
```
[the remark this answers is not in the document]
Comment (AdityaSripal) [reply modules/light-clients/06-solomachine/types/client_state.go:56]: Have you removed it from localhost?

**Review by AdityaSripal — APPROVED** [review]
LGTM. Check localhost has been changed correctly after Status PR.

File: modules/light-clients/06-solomachine/types/client_state.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/light-clients/06-solomachine/types/client_state.go:56]: yup, I think this one got added back during a merge of main when I fixed the solo machine bug

**Pull request merged**